### PR TITLE
[481380] Disallow $ character in project name

### PR DESCRIPTION
**Pull request by kthoms**
Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=481380

**Comment by kthoms** [comment]
screenshot 18

File: org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/WizardNewXtextProjectCreationPage.java [hunk 1]
```diff
@@ -228,6 +229,19 @@ protected boolean validatePage() {
...
 	}
 
+	private IStatus validateProjectName() {
+		String projectName = getProjectName();
+		IStatus status = JavaConventions.validatePackageName(projectName, JavaCore.VERSION_1_5, JavaCore.VERSION_1_5);
+		if (status.isOK()) {
+			// https://docs.oracle.com/javase/specs/jls/se8/html/jls-3.html#jls-3.8
+			// for historical reasons only underscore and $ are accepted as package name, but not for Xtext projects  
+			if (projectName.contains("$")) {
+				status = Status.CANCEL_STATUS;
```
Comment (spoenemann) [inline org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/WizardNewXtextProjectCreationPage.java]: Just a minor thing to be consistent: an ERROR status should be created here like it is done in `org.eclipse.jdt.core.JavaConventions#validatePackageName(String, String, String)`

Comment (kthoms) [inline org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/WizardNewXtextProjectCreationPage.java]: Updated. Although the actual status is not used further except for setting the error message. I'm not sure about this, the code became a bit more verbose without need.

Comment (spoenemann) [inline org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/WizardNewXtextProjectCreationPage.java]: Yes, but it was quite odd that the extracted method sometimes returned ERROR and sometimes CANCELED depending on the type of error.

**Pull request merged**